### DISCUSSION TITLE
Profile A vs. B

### DISCUSSION THREAD

**Discussion by Anuskuss** [quietvoid/hdr10plus_tool#82] [General]
What's the meaning behind those HDR10+ profiles? I've read multiple HDR10+ specification papers and it's not mentioned anywhere. Google has also let me down so far, except [this comment](https://www.liftgammagain.com/forum/index.php?threads/hdr10-in-resolve.10414/#post-137846) that said B is better than A (when exporting in DaVinci Resolve). Is this just a meaningless distinction (MediaInfo makes)?

Assuming that the comment is correct and B is a more advanced form of A, let's consider this: I have two 4K videos, one is A and the other is B.
![metadataA](https://github.com/user-attachments/assets/2c08ee3e-0954-43db-b3b1-7b5b1c594b00)
![metadataB](https://github.com/user-attachments/assets/808ebc34-49d9-4901-83b0-9d2d8689f917)

They look pretty similar but A has more scenes. Generally speaking, more scenes should be better, right? What other properties should be compared? A's metadata (177MB) is also noticeably smaller than B's (237MB). More scenes but less data. So which one is better?

<sub><sup>Not relevant to the discussion but it looks like the intern at the studio screwed up and put the credits on max brightness. Since it's white text on black background, let's hope you don't got a 10k nits display otherwise say goodbye to your eye sight.</sup></sub>

**Comment by quietvoid** [comment]
From the technical specification:

Profile A:
> HDR10+ Metadata for HDR10+ Profile A contains only the scene statistics information 

Profile B:
> HDR10+ Metadata for HDR10+ Profile B contains Basis Tone Mapping Curve (Bezier curve) information in
addition to scene statistics information

Basically profile A only has brightness metadata per scene (min/max/average as well as the brightness histogram).
Profile B defines a specific tone mapping curve per scene with a reference target.

Profile A was initially a request from some display manufacturers that wanted to implement their own tone mapping.

**Comment by Anuskuss** [comment]
So if B > A, should I inject B's metadata (better HDR10+) into A's video (higher bitrate)? Or are there other factors I should consider?

**Reply by quietvoid** [reply]
You need to confirm the videos are graded the same.
If they are, then just sync the metadata from one to the other.

**Reply by Anuskuss** [reply]
> videos are graded the same

I don't know what that means but they look the same and start at the same frame.

> If they are

The question was more regarding if it's more important to have more scenes or the extra Bezier curve.

**Reply by quietvoid** [reply]
The scenes number doesn't mean anything by itself.

**Reply by Anuskuss** [reply]
Oops, I just realized that they are not exactly the same (A has 4 pixels more at the top and at the bottom). Is there a way to add recenter it (not stretch) or should I just say f it at this point?

**Reply by quietvoid** [reply]
HDR10+ doesn't contain dimensions based metadata.

**Reply by Anuskuss** [reply]
Alright I thought "mapping" meant literally changing each pixel to a desired value but I guess not. So I'm good to go then? Thanks for the help!
